src/connection-manager/index.ts

```diff
@@ -427,9 +443,9 @@ export class DefaultConnectionManager extends EventEmitter<ConnectionManagerEven
     }
 
     const numConnections = this.getConnections().length
-    const toPrune = numConnections - this.opts.maxConnections
+    const toPrune = numConnections - this.opts.maxOutgoingConnections
```

Comment:
I feel like you should check `connection.direction` to check if incoming or outgoing so you can apply the appropriate limit. In theory, since you are enforcing `maxOutgoingConnections` in the `openConnection` function, you only need to handle incoming connections here.

Reply:
Good point, I would say better to keep them separate